### PR TITLE
add const

### PR DESCRIPTION
**Pull request by pavel-pimenov**

### CODE REVIEW

File: src/bt_peer_connection.cpp [hunk 1]
```diff
@@ -2026,7 +2026,7 @@ namespace libtorrent {
 		// for the metadata extension.
 		TORRENT_ASSERT(m_sent_handshake);
 
-		char msg[] = {0,0,0,0};
+        const char msg[] = {0,0,0,0};
```
Comment (aldenml) [inline src/bt_peer_connection.cpp]: indentation with spaces here, it should be `\t\t` (and 4 more below)

File: src/bt_peer_connection.cpp [hunk 2]
```diff
@@ -366,7 +366,7 @@ namespace libtorrent {
 #ifndef TORRENT_DISABLE_LOGGING
 		peer_log(peer_log_alert::outgoing_message, "HAVE_ALL");
 #endif
-		char msg[] = {0,0,0,1, msg_have_all};
+		const char msg[] = {0,0,0,1, msg_have_all};
```
Comment (arvidn) [inline src/bt_peer_connection.cpp]: for all the arrays used for messages, would you mind making them: ``static char const msg[]``?

Comment (pavel-pimenov) [inline src/bt_peer_connection.cpp]: ok

File: src/bt_peer_connection.cpp (hunk 1 above)
Comment (pavel-pimenov) [inline src/bt_peer_connection.cpp]: fixed. sorry

**Comment by pavel-pimenov** [comment]
@arvidn approve add private:
void send_buffer_internal(message_type const type, counters::stats_counter_t const counter);
?
default

**Comment by arvidn** [comment]
this PR looks good to me. the proposed refactor for sending single-byte messages also seem reasonable. Perhaps a better name would be something along the lines of: ``send_simple_message()`` (it should still be a private member though).

If you're up for it, I also think ``send_message()`` should take a ``span<char const>`` instead of a pointer and length.

**Comment by arvidn** [comment]
@pavel-pimenov do you intend to update this PR or should I merge this one?

**Comment by pavel-pimenov** [comment]
@arvidn i create separate PR(after merge). ok?

**Comment by pavel-pimenov** [comment]
PR for send_*_message https://github.com/arvidn/libtorrent/pull/1922

**Pull request merged**